packages/vite/src/plugins/ssr-styles.ts

```diff
@@ -93,10 +93,13 @@ export function ssrStylesPlugin (options: SSRStylePluginOptions): Plugin {
     async transform (code, id) {
       const { pathname, search } = parseURL(decodeURIComponent(pathToFileURL(id).href))
       const query = parseQuery(search)
-      if (!pathname.match(/\.(vue|((c|m)?j|t)sx?)$/g) || query.macro) { return }
-      if (options.shouldInline && !options.shouldInline(id)) { return }
 
       const relativeId = relativeToSrcDir(id)
+      if ((!pathname.endsWith('.server.vue') && !relativeId.startsWith('components/islands/'))) {
```

Comment:
can we have a better way of detecting whether these are island components, do you think? is this something the components module can handle for us?

Reply:
indeed, I was wondering if i should move it and use `getComponents` from `src/app/components/module`

Reply:
Or have some kind of way to detect it - maybe through listening to or calling a components hook?

Reply:
I think both are possible, within components module with `getComponents` and within vite with a `components:extend` 🤔 

do you prefer to keep it within `vite` or to move it to components  🙂 ?

Reply:
it looks nice ! i'll be waiting for it :+1:

Reply:
Done.

Reply:
https://github.com/nuxt/nuxt/blob/8b86d39e30ca970fb0adf2b5665a77f8dbbfb439/packages/nuxt/src/core/builder.ts#LL15C5-L15C5

Due to `app:templates` being called in `generateApp`, hooking into it won't work 🤔. to keep it within `nuxt/vite`, what do you think about passing down the `app` to `bundle()` ?
So we would have `bundle(nuxt, app)`  [here](https://github.com/nuxt/nuxt/blob/8b86d39e30ca970fb0adf2b5665a77f8dbbfb439/packages/nuxt/src/core/builder.ts#L39)

Reply:
Hm. Maybe instead we could set the resolved app on `nuxt.app` when we are building?

Reply:
ok - merged ✅